atomix/cluster/src/main/java/io/atomix/raft/storage/RaftStorage.java

```diff
@@ -110,17 +112,30 @@ public String prefix() {
    * @return indicates whether the lock was successfully acquired
    */
   public boolean lock(final String id) {
-    final File file = new File(directory, String.format(".%s.lock", prefix));
+    final File lockFile = new File(directory, String.format(".%s.lock", prefix));
+    final File tempLockFile = new File(directory, String.format(".%s.lock.tmp", id));
     try {
-      if (file.createNewFile()) {
-        Files.writeString(file.toPath(), id, StandardOpenOption.WRITE);
-        return true;
-      } else {
-        final String lock = Files.readString(file.toPath());
-        return lock != null && lock.equals(id);
+      if (!lockFile.exists()) {
+        // Create and update the file atomically
+        Files.writeString(
+            tempLockFile.toPath(),
+            id,
+            StandardOpenOption.CREATE,
+            StandardOpenOption.TRUNCATE_EXISTING,
+            StandardOpenOption.WRITE,
+            StandardOpenOption.SYNC);
+
+        // If two nodes tries to acquire lock, move will fail with FileAlreadyExistsException
+        FileUtil.moveDurably(
```

Comment:
:question: I can remember a discussion that this was only supported on some environments wasn't that something? Like only linux or? Is this an issue?

Reply:
Do you know what exactly was the problem? 

Flushing the parent directory does not work in windows. But according to what is documented in `FileUtil`, this is ok.

Reply:
We are using `FileUtil#moveDurably` in other places as well. So far no problems are reported. So I guess it should be ok. So I will merge this PR. If we see/know any problems later, let's tackle it then.